core/src/main/java/org/mskcc/cbio/portal/authentication/ldap/LDAPUserDetailsContextMapper.java

```diff
@@ -0,0 +1,217 @@
+
+import com.google.inject.internal.Lists;
+
+/**
+ * Mapping between details from an LDAP server and the user name used for authentication.
+ *
+ * Pull first name, given name, and email from the LDAP server.
+ *
+ * @author Manuel Holtgrewe <manuel.holtgrewe@bihealth.de>
+ */
+public class LDAPUserDetailsContextMapper implements UserDetailsContextMapper {
+
+    /** {@link Log} to use */
+    private static final Log LOG = LogFactory.getLog(LDAPUserDetailsContextMapper.class);
+
+    /** {@link DefaultSpringSecurityContextSource} with the LDAP connection */
+    private DefaultSpringSecurityContextSource ldapServer;
+    /** data access object to use for retrieving the authorities from the database */
+    private PortalUserDAO portalUserDAO;
+
+    /** base path to search users in */
+    private String baseDn;
+    /** attribute name to use for finding users in directory */
+    private String usernameAttribute = "sAMAccountName";
+    /** attribute name of the first / given name in directory */
+    private String givenNameAttribute = "givenName";
+    /** attribute name of the last name in the directory */
+    private String lastNameAttribute = "sn";
+    /** attribute name of the email address in the directory */
+    private String emailAttribute = "mail";
+
+    /** Default constructor */
+    public LDAPUserDetailsContextMapper() {
+    }
+
+    /**
+     * Initialize the {@link LDAPUserDetailsContextMapper} with the given {@link DefaultSpringSecurityContextSource} and
+     * {@link PortalUserDAO}.
+     */
+    public LDAPUserDetailsContextMapper(DefaultSpringSecurityContextSource ldapServer, PortalUserDAO portalUserDAO) {
+        this.ldapServer = ldapServer;
+        this.portalUserDAO = portalUserDAO;
+    }
+
+    @Override
+    public UserDetails mapUserFromContext(DirContextOperations ctx, String username,
+            Collection<? extends GrantedAuthority> authorities) {
+        // Query LDAP server for user details, including first/given name and email.
+        SpringSecurityLdapTemplate ldapTemplate = new SpringSecurityLdapTemplate(ldapServer);
+        String query = MessageFormat.format("({0}='{'0'}')", usernameAttribute);
+        DirContextOperations user = ldapTemplate.searchForSingleEntry(baseDn, query, new String[] { username });
+
+        // Extract attributes from the LDAP query result
+        String givenName = null;
+        if (user.getAttributeSortedStringSet(givenNameAttribute) != null)
+            givenName = (String) user.getAttributeSortedStringSet(givenNameAttribute).first();
+        String lastName = null;
+        if (user.getAttributeSortedStringSet(givenNameAttribute) != null)
+            lastName = (String) user.getAttributeSortedStringSet(givenNameAttribute).first();
+        String email;
+        if (user.getAttributeSortedStringSet(givenNameAttribute) == null) {
+            LOG.warn("Retrieving email failed, using username " + username + " for everything");
+            email = username;
```

Comment:
this will not work well with how the authorisation is currently documented (see https://github.com/cBioPortal/cbioportal/blob/master/docs/User-Authorization.md). The email should be there. 

When I mentioned that better exception handling would be nice, I meant only giving a clear exception message instead of NullPointerException. There is no need to implement such extra fallback scenarios (probably not desirable actually, as it complicates documentation).